src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/ConfigModelV7.java

```diff
@@ -936,38 +935,30 @@ private static String toQuotedCommaSeparatedString(final Set<String> roles) {
         }));
     }
 
-    private static boolean impliesTypePerm(Set<IndexPattern> ipatterns, Resolved resolved, User user, String[] actions,
-            IndexNameExpressionResolver resolver, ClusterService cs) {
-        Set<String> matchingIndex = new HashSet<>(resolved.getAllIndices());
-
-        for (String in : resolved.getAllIndices()) {
-            //find index patterns who are matching
-            Set<String> matchingActions = new HashSet<>(Arrays.asList(actions));
-            //Set<String> matchingTypes = new HashSet<>(resolved.getTypes(-));
-            for (IndexPattern p : ipatterns) {
-                if (WildcardMatcher.matchAny(p.getResolvedIndexPattern(user, resolver, cs), in)) {
-                    //per resolved index per pattern
-                    //for (String t : resolved.getTypes(-)) {
-                        //for (TypePerm tp : p.typePerms) {
-                            //if (WildcardMatcher.match(tp.typePattern, t)) {
-                                //matchingTypes.remove(t);
-                                for (String a : Arrays.asList(actions)) {
-                                    if (WildcardMatcher.matchAny(p.perms, a)) {
-                                        matchingActions.remove(a);
-                                    }
-                                }
-                            //}
-                        //}
-                    //}
-                }
-            }
-
-            if (matchingActions.isEmpty() /*&& matchingTypes.isEmpty()*/) {
-                matchingIndex.remove(in);
-            }
+    static final class IndexPatternsAndPermissions {
+        public String[] pattern;
+        public Set<String> perms;
+        public IndexPatternsAndPermissions(String[] pattern, Set<String> perms) {
+            this.pattern = pattern;
+            this.perms = perms;
         }
+    }
 
-        return matchingIndex.isEmpty();
+    private static boolean impliesTypePerm(Set<IndexPattern> ipatterns, Resolved resolved, User user, String[] requestedActions,
+                                           IndexNameExpressionResolver resolver, ClusterService cs) {
+        Set<String> resolvedRequestedIndices = resolved.getAllIndices();
+        List<IndexPatternsAndPermissions> indexPatternsAndPermissions = ipatterns
+                .stream()
+                .map(p -> new IndexPatternsAndPermissions(p.getResolvedIndexPattern(user, resolver, cs), p.perms))
+                .collect(Collectors.toList());
+        return resolvedRequestedIndices
+                .stream()
+                .allMatch(index ->
+                        Arrays.stream(requestedActions).allMatch(action -> indexPatternsAndPermissions
+                                .stream()
+                                .anyMatch(rip -> WildcardMatcher.matchAny(rip.pattern, index) && WildcardMatcher.matchAny(rip.perms, action))
```

Comment:
rename `rip` to `indexPatternsAndPermissions` (or less preferable `ipap`).

Reply:
used ipap as the scope is 3 lines and I believe in short names for short scopes.